File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    int[] numbers;
    int target;
    int N;
    public int solution(int[] numbers, int target) {
        N = numbers.length;
        this.numbers = numbers;
        this.target = target;
        dfs(0, 0);
        return answer;
    }
    void dfs(int i, int num) {
        if (i == N) {
            if (num == target) answer++;
            return;
        }
        dfs(i+1, num+numbers[i]);
        dfs(i+1, num-numbers[i]);
    }
}

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def solution(numbers, target):
    dfs(0, numbers, 0, target)
    return answer

def dfs(i, numbers, n, target):
    global answer
    if i == len(numbers):
        if n == target:
            answer += 1
        return
    
    dfs(i+1, numbers, n-numbers[i], target)
    dfs(i+1, numbers, n+numbers[i], target)